### DISCUSSION TITLE
how to automate updating files inside a torrent after they've been renamed?

### DISCUSSION THREAD

**Discussion by ghost** [transmission/transmission#6906] [Q&A] [answered]
Scenario. A torrent has been downloaded that contains 10 files. These files have names that are undesirable. A third-party tool (let's say filebot) is used to automatically rename these files.
Let's assume that transmission's settings.json has been configured to run a script after torrent completion. And that script is to run filebot which scans the 10 files and renames them according to whatever rules have been set up.

Now all of that is rather easily accomplished, but my question is: how might I automate updating the file names within transmission so I don't need to do hardlinks, symlinks, file moves, etc. The files can continue to seed as normal because they have been updated to the desired (new) names at both the filesystem level using filebot, and now also within the transmission-daemon.

The above step I have been doing manually via transmission-remote GUI, but I'd like to automate it. Any advice?

**Comment by tearfur** [comment]
You can use the RPC `torrent-rename-path` API to rename files instead, then Transmission will be updated too.

**Reply by ghost** [reply]
can you elaborate a bit more on this or link to some documentation?

**Reply by tearfur** [reply]
https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md

**Reply by tearfur** [reply, marked as answer]
You can use `transmission-remote`'s `--rename` option to achieve this as well, it will construct the RPC requests for you.

**Reply by botihu** [reply]
Here is an example how to remove space form the filename: `transmission-remote 192.168.2.3:9091 -n myuser:mypass --path 'folder/file name.ext' --rename "file-name.ext'`

**Comment by killemov** [comment]
Either go the path of having Transmission rename the files like mentioned above or rather copy/hardlink the the entire thing and then have filebot manipulate that.
```sh
cp -rupal source target
```
